src/handlers/StakeRequestHandler.ts

```diff
@@ -45,20 +60,25 @@ export default class StakeRequestHandler extends ContractEntityHandler<StakeRequ
    */
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   public async persist(transactions: any[]): Promise<StakeRequest[]> {
-    Logger.debug('Persisting stake request records');
-    const models: StakeRequest[] = transactions.map(
-      (transaction): StakeRequest => {
-        const { stakeRequestHash } = transaction;
-        const amount = new BigNumber(transaction.amount);
-        const beneficiary = Utils.toChecksumAddress(transaction.beneficiary);
-        const gasPrice = new BigNumber(transaction.gasPrice);
-        const gasLimit = new BigNumber(transaction.gasLimit);
-        const nonce = new BigNumber(transaction.nonce);
-        const gateway = Utils.toChecksumAddress(transaction.gateway);
-        const staker = Utils.toChecksumAddress(transaction.staker);
-        const stakerProxy = Utils.toChecksumAddress(transaction.stakerProxy);
+    const gatewayRecord = await this.gatewayRepository.getByChainGateway(
```

Comment:
if the intent here to verify this gateway exists in db, we should gracefully return if not found. `gatewayRecord!` might not be as readable. 
otherwise `gatewayAddress` was read from mosaic config and passed to this class, we can directly match `transaction.gateway` against it

Reply:
Changes done!